lib/matplotlib/_cm.py

```diff
@@ -1365,7 +1366,19 @@ def gfunc32(x):
     )
 
 
-datad = {
+class _deprecation_datad(dict):
+    def __getitem__(self, key):
+        if key in ["spectral", "spectral_r"]:
+            warn_deprecated(
+                "2.0",
+                name="spectral",
+                alternative="nipy_spectral",
+                obj_type="colormap"
+                )
```

Comment:
do you want to make the message dynamic for `spectral_r` as well ? I can see pros and cons to both.